### PR TITLE
AutoMez fix + pauseMelody integration

### PR DESCRIPTION
**Pull request by Trilkin**
[Advanced Settings.ini]
Added bard mez switch

[Bard.inc]
Fixed automez functionality (jedafix+trilkin)
Copied togglemez functionality from enchanter (jedafix)
Incorporated Sirhopsalot's pausemelody additions

### CODE REVIEW

File: Macros/e3 Includes/e3_Classes_Bard.inc [hunk 1]
```diff
@@ -114,7 +114,7 @@ SUB pauseTwist
 /if (${Debug} || ${Debug_Casting}) /echo |- pauseTwist ==>
 	/varset	returnTwist TRUE
 	/squelch /twist stop
-	/delay 20 !${Me.Casting.ID}
+	/delay 2 !${Me.Casting.ID}
```
Comment (sirhopsalot) [inline Macros/e3 Includes/e3_Classes_Bard.inc:117]: Why does this need to be changed from 2 seconds to .2 seconds? It's called in casting, loot, and buffcheck, so there might be side effects of this change.

File: Macros/e3 Includes/e3_Classes_Bard.inc [hunk 2]
```diff
@@ -125,6 +125,24 @@ SUB unpauseTwist
...
 /return
 |--------------------------------------------------------------|
+#event toggleMez "<#1#> Mez #2#"
+#event toggleMez "#1# tells you, 'Mez #2#'"
+sub event_toggleMez(line, ChatSender, mezToggle)
+  /if (${Me.Class.ShortName.Equal[ENC]} || ${Me.Class.ShortName.Equal[BRD]}) {
+    /if (${Debug} || ${Debug_Ench}) /echo |- remove_RuneTarget ==>
+      /if (${mezToggle.Equal[on]}) {
+        /varset autoMez TRUE
+        /varset fixedMelody ${currentMelody}
```
Comment (sirhopsalot) [inline Macros/e3 Includes/e3_Classes_Bard.inc:135]: Just want to understand. Why does mezzing require a fixedMelody? Will changing melodies mess with autoMez?

File: Macros/e3 Includes/e3_Classes_Bard.inc [hunk 3]
```diff
@@ -202,21 +220,24 @@ sub check_bard_mez
...
+      /if (${Select[${AssistTarget},${mezTarg}]}==0 && (!${Bool[${Target.Mezzed}]} || ${Target.Mezzed.Duration.TotalSeconds} < 7)) {
         /for s 1 to ${mezArray2D.Size[1]}
           /if (${check_Distance[${mezTarg},${mezArray2D[${s},${iMyRange}]}]}) {
             /if (${mezArray2D[${s},${iCastType}].Equal[Spell]} && !${Bool[${Me.Gem[${mezArray2D[${s},${iCastName}]}]}]}) /call memorize_spell "${mezArray2D[${s},${iCastName}]}" ${mezArray2D[${s},${iSpellGem}]}
               /docommand ${ChatToggle} << Mezzing ${mezTarg} with ${mezArray2D[${s},${iCastName}]} >>
               /if (${Twist.Twisting}) /call pauseTwist
               |:retryMez
               /twist once ${mezArray2D[${s},${iSpellGem}]}
-              /delay 3s !${Me.Casting.ID}
+              /delay 3s
```
Comment (sirhopsalot) [inline Macros/e3 Includes/e3_Classes_Bard.inc:241]: The !${Me.Casting.ID} should cause it to break out early if it finishes casting. Is that a problem for some reason?

**Review by sirhopsalot — CHANGES_REQUESTED** [review]

**Comment by Trilkin** [comment]
> The !${Me.Casting.ID} should cause it to break out early if it finishes casting. Is that a problem for some reason? 

Yes. It catches it in a loop where it repeatedly stops the mez and tries to restart it.

**Comment by Trilkin** [comment]
> Just want to understand. Why does mezzing require a fixedMelody? Will changing melodies mess with autoMez? 

Yes. If a melodyIf is triggered, the bard will try to change melodies first.

**Comment by Trilkin** [comment]
> Why does this need to be changed from 2 seconds to .2 seconds? It's called in casting, loot, and buffcheck, so there might be side effects of this change. 

I think this is a typo from when I merged it from the first round of changes. It's supposed to be 2s for consistency.

**Comment by Trilkin** [comment]
> > Why does this need to be changed from 2 seconds to .2 seconds? It's called in casting, loot, and buffcheck, so there might be side effects of this change. 

I think this is a typo from when I merged it from the first round of changes. It's supposed to be 2s for consistency. 

Never mind, this isn't a typo. This is to keep it in line with the behavior of unpause twist.

File: Macros/e3 Includes/e3_Classes_Bard.inc (hunk 2 above)
Comment (sirhopsalot) [reply Macros/e3 Includes/e3_Classes_Bard.inc:135]: That seems fine

File: Macros/e3 Includes/e3_Classes_Bard.inc (hunk 3 above)
Comment (sirhopsalot) [reply Macros/e3 Includes/e3_Classes_Bard.inc:241]: Very odd

**Review by sirhopsalot — APPROVED** [review]

**Pull request merged**